x-pack/plugins/maps/public/connected_components/timeslider/timeslider.tsx

```diff
@@ -0,0 +1,179 @@
+    this._onChange([this.state.ticks[0].value, this.state.ticks[1].value]);
+  }
+
+  _doesTimesliceCoverTimerange() {
+    return this.state.timeslice[0] === this.state.min && this.state.timeslice[1] === this.state.max;
+  }
+
+  _onDualControlChange = (value: [number | string, number | string]) => {
+    this.setState({ range: (value[1] as number) - (value[0] as number) }, () => {
+      this._onChange(value as [number, number]);
+    });
+  };
+
+  _onChange = (value: [number, number]) => {
+    this.setState({
+      timeslice: value,
+      timesliceText: prettyPrintTimeslice(value),
+    });
+    this._propagateChange(value);
+  };
+
+  _onNext = () => {
+    const from =
+      this._doesTimesliceCoverTimerange() || this.state.timeslice[1] === this.state.max
+        ? this.state.ticks[0].value
+        : this.state.timeslice[1];
+    const to = from + this.state.range;
+    this._onChange([from, to <= this.state.max ? to : this.state.max]);
+  };
+
+  _onPrevious = () => {
+    const to =
+      this._doesTimesliceCoverTimerange() || this.state.timeslice[0] === this.state.min
+        ? this.state.ticks[this.state.ticks.length - 1].value
+        : this.state.timeslice[0];
+    const from = to - this.state.range;
+    this._onChange([from < this.state.min ? this.state.min : from, to]);
+  };
+
+  _propagateChange = _.debounce((value: [number, number]) => {
+    if (this._isMounted) {
+      this.props.setTimeslice({ from: value[0], to: value[1] });
+    }
+  }, 300);
+
+  render() {
+    return (
+      <div className="mapTimeslider">
+        <div className="mapTimeslider__row">
+          <EuiButtonIcon
+            onClick={this.props.closeTimeslider}
+            iconType="cross"
+            color="subdued"
+            className="mapTimeslider__close"
+            aria-label={i18n.translate('xpack.maps.timeslider.closeLabel', {
+              defaultMessage: 'Close timeslider',
+            })}
+          />
+
+          <div className="mapTimeslider__timeWindow">
+            <EuiText size="s">{this.state.timesliceText}</EuiText>
+          </div>
+
+          <div className="mapTimeslider__innerPanel">
+            <div className="mapTimeslider__controls">
+              <EuiButtonIcon
+                onClick={this._onPrevious}
+                iconType="arrowLeft"
+                color="text"
+                aria-label={i18n.translate('xpack.maps.timeslider.previousTimeWindowLabel', {
+                  defaultMessage: 'Previous time window',
+                })}
+              />
+              <EuiButtonIcon
+                onClick={this._onNext}
+                iconType="arrowRight"
+                color="text"
+                aria-label={i18n.translate('xpack.maps.timeslider.nextTimeWindowLabel', {
+                  defaultMessage: 'Next time window',
+                })}
+              />
+            </div>
+          </div>
+        </div>
+
+        <div className="mapTimeslider__row">
+          <EuiDualRange
```

Comment:
I guess these are more EuiDualRange comments:

is it possible to drag by the actual bar (iso of just the endpoints)?

`onChange` can fire while the user is still manipulating (ie. "holding an endpoint"). it is possible to only fire _after_ the user has let go.  Right now, we're getting refreshes while the user is still trying to move the slider.

Perhaps we can fix this higher up in here. Is it easy to detect where the mouse-is pressed?

cc @miukimiu

Reply:
> is it possible to drag by the actual bar (iso of just the endpoints)?

PR is out for review in EUI to add this functionality, https://github.com/elastic/eui/pull/4776

> onChange can fire while the user is still manipulating (ie. "holding an endpoint"). it is possible to only fire after the user has let go. Right now, we're getting refreshes while the user is still trying to move the slider.

Would you mind creating an issue in EUI for this?

Reply:
https://github.com/elastic/eui/issues/4812